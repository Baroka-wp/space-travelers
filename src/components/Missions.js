import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../redux/missions/missionsReducer';
import MissionCard from './pages/MissionCard';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissions());
  }, []);

  return (
    <div className="container">
      <table className="table table-striped table-bordered">
        <thead className="table-mission-head">
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th> </th>
        </thead>
        <tbody>
          {
              missions.map((mission) => (
                <MissionCard key={mission.mission_id} mission={mission} />
              ))
            }
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
