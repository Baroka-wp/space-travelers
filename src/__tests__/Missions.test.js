import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/Missions';

describe('Mission Element Tests', () => {
  test('if Missions Component Renders in DOM porperly', () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(screen.queryByText(/Rockets/)).toBeNull();
  });
  it('renders porperly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
