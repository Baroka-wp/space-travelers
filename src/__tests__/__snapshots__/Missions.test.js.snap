// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Element Tests renders porperly 1`] = `
<div
  className="container"
>
  <table
    className="table table-striped table-bordered"
  >
    <thead>
      <tr
        className="table-mission-head"
      >
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
